src/core/tokenizer.ts

```diff
@@ -109,24 +91,25 @@ function generateMatcher(
   };
 }
 
-const languageTokenizerOverrides: Partial<
-  Record<LanguageId, LanguageTokenizerOverrides>
-> = {
-  css: languageWithDashedIdentifiers,
-  scss: languageWithDashedIdentifiers,
-  shellscript: languageWithDashedIdentifiers,
-};
-
-const tokenMatchersForLanguage: Partial<Record<LanguageId, Matcher>> =
-  mapValues(languageTokenizerOverrides, (val: LanguageTokenizerComponents) =>
-    generateMatcher(val)
-  );
+const matchers = new Map<string, Matcher>();
 
 export function getMatcher(languageId: string): Matcher {
-  return (
-    tokenMatchersForLanguage[languageId as SupportedLanguageId] ??
-    defaultMatcher
-  );
+  const wordSeparators = tokenizerConfiguration.getWordSeparators(languageId);
+  const key = wordSeparators.join();
```

Comment:
This is going to result in subtle bugs in certain extremely rare cases.  Eg `["--", "_"]` and `["-", "-_"]` will map to the same key.  I know it's extremely unlikely that this will happen, but do we really want to knowingly introduce a subtle bug that will probably be really confusing for that one user that hits it?

Reply:
Note that if you're worried about performance from the json stringify, we can do more effective caching once we switch to graph impl, as then we can just cache the language's regex and listen for settings changes directly to invalidate it

Reply:
No it still joins with a `,`

Reply:
That still has potential for bugs, though they're even rarer.  How about

```suggestion
  const key = wordSeparators.join("\u0000");
```

Reply:
I guess it's fine but the chances that someone's going to have a comma in the middle of an identifier must be close to zero.